Sources/Snowplow/Events/EventBase.swift

```diff
@@ -82,18 +82,18 @@ public class Event: NSObject {
         return self
     }
     
-    /// Replace the context entities attached to the event with a new list of entities.
+    /// Adds a list of context entities to the existing ones.
     @objc
     public func entities(_ entities: [SelfDescribingJson]) -> Self {
-        self.entities = entities
+        self.entities.append(contentsOf: entities)
```

Comment:
An edge case, but I think this would work better, even if `isProcessing` is true.

```suggestion
        self._entities.append(contentsOf: entities)
```